### ISSUE TITLE
Error in reverse_bits()  method when used with circuits with isolated qubits

### ISSUE THREAD

**Issue by TharrmashasthaPV** [Qiskit/qiskit-terra#6731]
### Information

- **Qiskit Terra version**: 0.18
- **Python version**: 3.7

### What is the current behavior?

This follows #6574 . When a `QuantumCircuit` is initialized with isolates qubits there are some issues as follows

- [ ] `qc.reverse_bits()` returns an empty `QuantumCircuit` object with no qubits or clbits when `qc` has no gates.
- [ ] `qc.reverse_bits()` returns `list index out of range` error when `qc` has atleast 1 gate. (from #6574 )

### Steps to reproduce the problem

1.
```python
from qiskit import *
from qiskit.circuit import Qubit

qc = QuantumCircuit([Qubit(), Qubit()])
qc_rev = qc.reverse_bits()
print(qc_rev.qubits)  # Returns []
```

2.
```python
from qiskit import *
from qiskit.circuit import Qubit

qc = QuantumCircuit([Qubit(), Qubit()])
qc.h(0)
qc_rev = qc.reverse_bits()  # Returns `list index out of range`
```

**Comment by apcarrik** [comment]
I'd be happy to work on this if no one else is.

**Comment by TharrmashasthaPV** [comment]
@apcarrik Yeah sure please go ahead and please feel free to leave a message if you need some help.😃

**Comment by apcarrik** [comment]
Thanks @TharrmashasthaPV! For clarification, is the intent of passing a list of Qubits as an argument to `QuantumCircuit()` to create a `QuantumRegister` implicitly with them? Currently, it updates the `QuantumCircuit._qubits` and `QuantumCircuit._qubits_set` properties, but does not create a quantum register in the quantum circuit, which causes the issues in `QuantumCircuit.reverse_bits()` method that we're seeing.

**Comment by TharrmashasthaPV** [comment]
> Thanks @TharrmashasthaPV! For clarification, is the intent of passing a list of Qubits as an argument to `QuantumCircuit()` to create a `QuantumRegister` implicitly with them? Currently, it updates the `QuantumCircuit._qubits` and `QuantumCircuit._qubits_set` properties, but does not create a quantum register in the quantum circuit, which causes the issues in `QuantumCircuit.reverse_bits()` method that we're seeing.

@apcarrik Not exactly. The input to the circuit is a set of registerless qubits. And so we would like to keep the qubits that way without creating a quantumregister. I think currently `reverse_bits()` returns a circuit that only contains reversed registers. However, we need to fix it to also contain the reversed registerless qubits and registerless clbits without creating registers for them.

**Comment by apcarrik** [comment]
> > Thanks @TharrmashasthaPV! For clarification, is the intent of passing a list of Qubits as an argument to `QuantumCircuit()` to create a `QuantumRegister` implicitly with them? Currently, it updates the `QuantumCircuit._qubits` and `QuantumCircuit._qubits_set` properties, but does not create a quantum register in the quantum circuit, which causes the issues in `QuantumCircuit.reverse_bits()` method that we're seeing.
> 
> @apcarrik Not exactly. The input to the circuit is a set of registerless qubits. And so we would like to keep the qubits that way without creating a quantumregister. I think currently `reverse_bits()` returns a circuit that only contains reversed registers. However, we need to fix it to also contain the reversed registerless qubits and registerless clbits without creating registers for them.

Ok, thanks for clarifying that.

**Comment by Gaya3-mv** [comment]
@apcarrik Is the issue fixed? Or else, I would like to give it a try.

**Comment by apcarrik** [comment]
> @apcarrik Is the issue fixed? Or else, I would like to give it a try.

There was a lengthy discussion on what exactly the desired behavior is for this, which never really came to a consensus (see PR #[6762](https://github.com/Qiskit/qiskit-terra/pull/6762).

**Comment by Gaya3-mv** [comment]
@apcarrik Ok. Sorry, I didn't know about all this progress regarding fixing this issue. Thanks for updating that you are continuing on fixing it.

**Comment by apcarrik** [comment]
@Gaya3-mv No need to apologize! I just wanted to let you know what was discussed on the PR. I'm going to close the PR since it's stale, and it seems like #7628 addresses the broader issue of treating isolated qubits and qregs similarly. Feel free to pick this issue or #7628 up!

**Comment by Gaya3-mv** [comment]
Ok. Thanks again for the details regarding the status of the issue!

**Comment by RcrdPhysics** [comment]
Hi, is this issue open and not assigned, could I work on this issue?

**Comment by apcarrik** [comment]
> Hi, is this issue open and not assigned, could I work on this issue?

Feel free to pick it up.

**Comment by srivarenya01** [comment]
@apcarrik  Is this issue still open? 
CC: @RcrdPhysics

**Comment by 1ucian0** [comment]
This issue is not reproducible anymore:

---
```python
from qiskit import *
from qiskit.circuit import Qubit

qc = QuantumCircuit([Qubit(), Qubit()])
qc_rev = qc.reverse_bits()
print(qc_rev.qubits) 
```
```
[<Qubit uid=63>, <Qubit uid=62>]
```
---
```python
from qiskit import *
from qiskit.circuit import Qubit

qc = QuantumCircuit([Qubit(), Qubit()])
qc.h(0)
qc_rev = qc.reverse_bits()
print(qc_rev.qubits)
```
```
[<Qubit uid=67>, <Qubit uid=66>]
```
